GeneratorInterface/GenFilters/plugins/LHEPtFilter.cc

```diff
@@ -87,19 +91,39 @@ bool LHEPtFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSetu
     }
   }
   double vpt_ = -1;
-  if (!cands.empty()) {
-    ROOT::Math::PxPyPzEVector tot = cands.at(0);
-    for (unsigned icand = 1; icand < cands.size(); ++icand) {
-      tot += cands.at(icand);
+  if (isScalar_) {  // do the scalar sum
+    if (!cands.empty()) {
+      double tot = cands.at(0).pt();
+      for (unsigned icand = 1; icand < cands.size(); ++icand) {
+        tot += cands.at(icand).pt();
+      }
+      vpt_ = tot;
+    }
+    if ((ptMax_ < 0. || vpt_ <= ptMax_) && vpt_ > ptMin_) {
+      return true;
+    } else {
+      return false;
+    }
+  } else {  // else do the vector sum
+    if (!cands.empty()) {
+      ROOT::Math::PxPyPzEVector tot = cands.at(0);
+      for (unsigned icand = 1; icand < cands.size(); ++icand) {
+        tot += cands.at(icand);
+      }
+      vpt_ = tot.pt();
+    }
+    if ((ptMax_ < 0. || vpt_ <= ptMax_) && vpt_ > ptMin_) {
+      return true;
+    } else {
+      return false;
     }
-    vpt_ = tot.pt();
-  }
-  if ((ptMax_ < 0. || vpt_ <= ptMax_) && vpt_ > ptMin_) {
-    return true;
-  } else {
-    return false;
   }
 }
 
+void LHEPtFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<bool>("isScalar", false);  // default is false
+}
```

Comment:
> The default behavior is preserved via fillDescriptions.

Unfortunately this isn't really the case, because `desc` is not added to the `descriptions`, and therefore this `fillDescriptions()` function is a no-op. Minimal fix would be
```cpp
  descriptions.addDefault(desc);
```
For more information see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

Reply:
Thanks for the catch Matti, I will fix this in a follow up PR.